yellowbox/extras/logstash.py

```diff
@@ -1,30 +1,263 @@
-from typing import Optional
-
-from docker import DockerClient
+import json
+import logging
+import selectors
+import socket
+import threading
+from typing import Any, Callable, Dict, Iterable, List, Optional, Union, cast, TypedDict
+from weakref import WeakMethod
 
 from yellowbox.retry import RetrySpec
-from yellowbox.containers import get_ports, create_and_pull
-from yellowbox.subclasses import SingleContainerService, RunMixin
+from yellowbox.subclasses import YellowService
+from yellowbox.utils import _docker_host_name
+
+__all__ = ['LogstashService']
+_logger = logging.getLogger(__name__)
+
+_LISTEN_BACKLOG = 5  # Awaiting connections backlog count
```

Comment:
this should definitely be an init arg

Reply:
What why? You'll never change this and it doesn't matter at all if we don't cap the connections.

Reply:
If a blackbox tests has more than 5 logging components, this may cause issues

Reply:
No it doesn't. The server can have a 100 connections.

Reply:
standard is 128 AFAIK

Reply:
It's a backlog ben, it doesn't matter what value you put in there if you don't cap the simultaneous connections.

Reply:
Hey, ever since 3.5 this parameter is optional. Thank god. I'm just removing this and letting Python choose a "reasonable unknown value that I don't care about".